### PR TITLE
Issue #323: Adding support for activity codes for Phoenix files

### PR DESCRIPTION
**Pull request by rsinha9**

### CODE REVIEW

File: src/main/java/net/sf/mpxj/phoenix/Phoenix5Reader.java [hunk 1]
```diff
@@ -451,15 +487,28 @@ private void processActivities(Storepoint phoenixProject)
...
+
       task.setActivityID(activity.getId());
 
       task.setActualDuration(activity.getActualDuration());
       task.setActualFinish(activity.getActualFinish());
       task.setActualStart(activity.getActualStart());
       //activity.getBaseunit()
       //activity.getBilled()
-      //activity.getCalendar()
+      task.setCalendar(m_projectFile.getCalendarByName(activity.getCalendar()));
```
Comment (joniles) [inline src/main/java/net/sf/mpxj/phoenix/Phoenix5Reader.java:511]: The updates to set the calendar seem unrelated to the activity code changes, can we create a separate PR for these? It's easier for me to see the changes when I run my test suite with the modified code.

File: src/main/java/net/sf/mpxj/ActivityCode.java [hunk 2]
```diff
@@ -132,10 +146,17 @@ public List<ActivityCodeValue> getValues()
...
-   private final Integer m_sequenceNumber;
-   private final String m_name;
+   @Override public int compareTo(ActivityCode o)
+   {
+      int id1 = NumberHelper.getInt(getUniqueID());
+      int id2 = NumberHelper.getInt(o.getUniqueID());
+      return (Integer.compare(id1, id2));
+   }
+
+   private Integer m_uniqueID;
```
Comment (joniles) [inline src/main/java/net/sf/mpxj/ActivityCode.java:156]: General question: the PR switches a couple of classes from being immutable to mutable. I was just wondering if there was a specific reason for this? I appreciate that the MPXJ codebase is not consistent in this respect (20 years of legacy code!) but it's something I think about when I'm adding new classes. Interested to hear you thoughts.

File: src/main/java/net/sf/mpxj/phoenix/Phoenix5Reader.java (hunk 1 above)
Comment (rsinha9) [reply src/main/java/net/sf/mpxj/phoenix/Phoenix5Reader.java:511]: Yup. We will split this into 2 PRs.

File: src/main/java/net/sf/mpxj/ActivityCode.java (hunk 2 above)
Comment (rsinha9) [reply src/main/java/net/sf/mpxj/ActivityCode.java:156]: I prefer having data classes be immutable, we changed this to mimic the same pattern that was being used for Task and Resources. We'd be happy to update this to be immutable if that is the path we want to go down. I know for Tasks that would be difficult because of the number of parameters. Typically when I have a large number of parameters I like to employ a builder pattern. If that's something you are open to, we can do that here and then over time get the others to mimic the same pattern.
